### PR TITLE
ch4/ofi: move am inflight trackers into per_vni

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
We need track and progress individual vni's am inflight trackers,
especially during finalize.

Fixes pmodels/mpich#5627

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_types.h [hunk 1]
```diff
@@ -235,6 +235,10 @@ typedef struct {
     void *am_send_seq_tracker;
     void *am_recv_seq_tracker;
 
+    /* active message trackers */
+    int am_inflight_inject_emus;
+    int am_inflight_rma_send_mrs;
+
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_types.h:241]: I guess these are always modified under lock protection? So the atomic types are no longer needed?

Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_types.h:241]: Yes. Earlier I made them into atomic types so they are shared across all vnis. But moving them into per-vni is much cleaner.

**Review by raffenet — APPROVED** [review]

**Pull request merged**